builtin/logical/pki/crl_util.go

```diff
@@ -26,6 +30,65 @@ type revocationInfo struct {
 	CertificateIssuer issuerID  `json:"issuer_id"`
 }
 
+// crlBuilder is gatekeeper for controlling various read/write operations to the storage of the CRL.
+// The extra complexity arises from secondary performance clusters seeing various writes to its storage
+// without the actual API calls. During the storage invalidation process, we do not have the required state
+// to actually rebuild the CRLs, so we need to schedule it in a deferred fashion. This allows either
+// read or write calls to perform the operation if required, or have the flag reset upon a write operation
+type crlBuilder struct {
+	m            sync.Mutex
+	forceRebuild uint32
+}
+
+const (
+	_ignoreForceFlag  = true
+	_enforceForceFlag = false
+)
+
+// rebuildIfForced is to be called by readers or periodic functions that might need to trigger
+// a refresh of the CRL before the read occurs.
+func (cb *crlBuilder) rebuildIfForced(ctx context.Context, b *backend, request *logical.Request) error {
+	if atomic.LoadUint32(&cb.forceRebuild) == 1 {
+		return cb._doRebuild(ctx, b, request, true, _enforceForceFlag)
+	}
+
+	return nil
+}
+
+// rebuild is to be called by various write apis that know the CRL is to be updated and can be now.
+func (cb *crlBuilder) rebuild(ctx context.Context, b *backend, request *logical.Request, forceNew bool) error {
+	return cb._doRebuild(ctx, b, request, forceNew, _ignoreForceFlag)
+}
+
+// requestRebuildOnActiveNode will schedule a rebuild of the CRL from the next read or write api call assuming we are the active node of a cluster
+func (cb *crlBuilder) requestRebuildOnActiveNode(b *backend) {
```

Comment:
Can we call this requestRebuild*If*ActiveNode?  I initially thought this somehow sent a request to the active node.

Reply:
Yup much better name.